### PR TITLE
[Merged by Bors] - chore: add missing #align statements

### PR DESCRIPTION
**Pull request by jcommelin**
This PR is the result of a slight variant on the following "algorithm"

* take all mathlib 3 names, remove `_` and make all uppercase letters into lowercase
* take all mathlib 4 names, remove `_` and make all uppercase letters into lowercase
* look for matches, and create pairs `(original_lean3_name, OriginalLean4Name)`
* for pairs that do not have an align statement:
  - use Lean 4 to lookup the file + position of the Lean 4 name
  - add an `#align` statement just before the next empty line
* manually fix some tiny mistakes (e.g., empty lines in proofs might cause the `#align` statement to have been inserted too early)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by kbuzzard — COMMENTED** [review]
I have actually looked at all 222 changed files here; I left very few comments on specific additions. I think this is a great PR which presumably will be rotting very quickly; if I had merge powers I'd almost be tempted to merge now and fix up the potential issues I flagged later. 

The PR seems to have caught the fact that in four or five files, nothing is aligned at all! Many additions are "null" in the sense that they're aligning xyz with xyz but there are still a huge number of genuine alignments.

I've always been aware that mathlib3port was adapted to deal with to_additive aligns some time after to_additive was ported, so there would be some to_additive stuff which had been missed. The technique used in this PR has caught these, but it has also caught declarations which have been generated by `ext`, `simps` and `reassoc`. It is probably worth opening an issue in mathlib3port asking that these declarations are auto-aligned by the porting program, otherwise you'll have to run this code again and again as the port progresses. Once the porting program knows about these (and it might do so already, I'm just observing that clearly it didn't always know about these things) you might just need to run it one last time. 

Excellent work! Let's get this merged quickly.

File: Mathlib/CategoryTheory/ConcreteCategory/Bundled.lean [hunk 1]
```diff
@@ -10,6 +10,7 @@ Authors: Scott Morrison, Johannes Hölzl, Reid Barton, Sean Leather
 -/
 import Std.Tactic.Lint.Frontend
 import Std.Tactic.Lint.Misc
+import Mathlib.Mathport.Rename
```
[the remark this answers is not in the document]
Comment (jcommelin) [reply Mathlib/CategoryTheory/ConcreteCategory/Bundled.lean:13]: Yes, it provides the `#align` command.

File: Mathlib/Data/Fin/Fin2.lean [hunk 2]
```diff
@@ -10,6 +10,7 @@ Authors: Mario Carneiro
 -/
 import Std.Tactic.NoMatch
 import Mathlib.Init.Data.Nat.Notation
+import Mathlib.Mathport.Rename
```
[the remark this answers is not in the document]
Comment (jcommelin) [reply Mathlib/Data/Fin/Fin2.lean:13]: Ditto here

File: Mathlib/Data/Int/Cast/Basic.lean [hunk 3]
```diff
@@ -68,6 +68,7 @@ theorem cast_ofNat (n : ℕ) : ((n : ℤ) : R) = n :=
   AddGroupWithOne.intCast_ofNat _
 #align int.cast_coe_nat Int.cast_ofNatₓ
 -- expected `n` to be implicit, and `HasLiftT`
+#align int.cast_of_nat Int.cast_ofNat
```
[the remark this answers is not in the document]
Comment (jcommelin) [reply Mathlib/Data/Int/Cast/Basic.lean:71]:
```suggestion
#align int.cast_of_nat Int.cast_ofNatₓ
```

File: Mathlib/Algebra/BigOperators/Pi.lean [hunk 4]
```diff
@@ -71,6 +71,7 @@ theorem prod_mk_prod {α β γ : Type _} [CommMonoid α] [CommMonoid β] (s : Fi
   Finset.induction_on s rfl (by simp (config := { contextual := true }) [Prod.ext_iff])
 #align prod_mk_prod prod_mk_prod
 #align sum_mk_sum prod_mk_sum
```
Comment (jcommelin) [inline Mathlib/Algebra/BigOperators/Pi.lean:73]:
```suggestion
```

File: Mathlib/Algebra/BigOperators/Pi.lean [hunk 5]
```diff
@@ -71,6 +71,7 @@ theorem prod_mk_prod {α β γ : Type _} [CommMonoid α] [CommMonoid β] (s : Fi
...
+#align prod_mk_sum prod_mk_sum
```
[the remark this answers is not in the document]
Comment (jcommelin) [reply Mathlib/Algebra/BigOperators/Pi.lean:73]: `sum_mk_sum` doesn't even exist in mathlib3 :grinning:

**Comment by fpvandoorn** [comment]
bors merge

**Pull request closed without merging**